src/psyclone/transformations.py

```diff
@@ -2860,7 +2860,7 @@ def _validate(self, sched):
                 "a schedule of type {0}".format(type(sched)))
 
         # Check that we don't already have a data region of any sort
-        directives = sched.walk(sched.children, Directive)
+        directives = sched.walk(Directive)
         data_directives = [True if isinstance(ddir, (ACCDataDirective,
```

Comment:
Again, not yours but pylint says this line can be simplified to something like `data_directives = [bool(isinstance(ddir, (...)) for ddir in directives]`. You might not need the `bool()` constructor though.

Reply:
I simplified this a lot further ... using 'any' will allow short-circuit :)